### PR TITLE
Updated eslint rules for ECMAScript 7

### PR DESCRIPTION
**Pull request by JoergAtGithub**

### CODE REVIEW

**Review by Be-ing — APPROVED** [review]

File: .eslintrc.json [hunk 1]
```diff
@@ -1,9 +1,12 @@
 {
   "extends": "eslint:recommended",
   "parserOptions": {
-    "ecmaVersion": 5,
+    "ecmaVersion": 7,
```
Comment (Holzhaus) [inline .eslintrc.json:4]: I think ES7 support is incomplete before 5.12 (?), so we should stick with ES6 for now, until our minimum Qt version supports it.

Comment (Holzhaus) [reply .eslintrc.json:4]: https://bugreports.qt.io/browse/QTBUG-47735

Comment (Holzhaus) [reply .eslintrc.json:4]: If someone uses Ubuntu 18.04 (Qt 5.9) there would be weird errors in controller mappings that don't happen on other systems. I think we should stick with ES3 until Ubuntu 20.04 (Qt 5.12) becomes the minimum requirement.

Comment (Be-ing) [reply .eslintrc.json:4]: I'm quite frustrated that we're even talking about this anymore. Ubuntu 18.04 is 3 years old and has been outdated by another Ubuntu LTS for a year now. Let's just raise the minimum requirement and be done with it.

Comment (Holzhaus) [reply .eslintrc.json:4]: I agree, but we should stick with the requirements policy decided as a team.

I looked it up:

> The main branch must build with the packages in the latest Ubuntu LTS which has had at least one point release.
https://github.com/mixxxdj/mixxx/wiki/Minimum%20requirements%20policy

**Ubuntu 20.04**

LTS? Yes ✅
At least one point release? Yes, Ubuntu 20.04.2 is already released ✅

So I guess it's okay to require at least Qt 5.12 for Mixxx 2.4.

Comment (uklotzde) [reply .eslintrc.json:4]: I agree. Qt 5.12 should be the minimum for 2.4!

**Review by Holzhaus — APPROVED** [review]

**Pull request merged**